metadata/src/test/java/org/apache/kafka/metadata/migration/KRaftMigrationDriverTest.java

```diff
@@ -0,0 +1,315 @@
+        return record;
+    }
+
+    /**
+     * Enqueues a metadata change event with the migration driver and returns a future that can be waited on in
+     * the test code. The future will complete once the metadata change event executes completely.
+     */
+    CompletableFuture<Void> enqueueMetadataChangeEventWithFuture(
+        KRaftMigrationDriver driver,
+        MetadataDelta delta,
+        MetadataImage newImage,
+        MetadataProvenance provenance
+    ) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        Consumer<Throwable> completionHandler = ex -> {
+            if (ex == null) {
+                future.complete(null);
+            } else {
+                future.completeExceptionally(ex);
+            }
+        };
+
+        driver.enqueueMetadataChangeEvent(delta, newImage, provenance, false, completionHandler);
+        return future;
+    }
+
+    @Test
+    public void testOnlySendNeededRPCsToBrokers() throws Exception {
+        // KAFKA-14668 Don't send RPCs to brokers for every metadata change, only when broker or topics change.
+        CountingMetadataPropagator metadataPropagator = new CountingMetadataPropagator();
+        CapturingMigrationClient migrationClient = new CapturingMigrationClient(new HashSet<>(Arrays.asList(1, 2, 3)));
+        KRaftMigrationDriver driver = new KRaftMigrationDriver(
+            3000,
+            new NoOpRecordConsumer(),
+            migrationClient,
+            metadataPropagator,
+            metadataPublisher -> { },
+            new MockFaultHandler("test")
+        );
+
+        MetadataImage image = MetadataImage.EMPTY;
+        MetadataDelta delta = new MetadataDelta(image);
+
+        driver.start();
+        delta.replay(zkBrokerRecord(1));
+        delta.replay(zkBrokerRecord(2));
+        delta.replay(zkBrokerRecord(3));
+        MetadataProvenance provenance = new MetadataProvenance(100, 1, 1);
+        image = delta.apply(provenance);
+
+        // Publish a delta with this node (3000) as the leader
+        driver.publishLogDelta(delta, image, new LogDeltaManifest(provenance,
+            new LeaderAndEpoch(OptionalInt.of(3000), 1), 1, 100, 42));
+
+        TestUtils.waitForCondition(() -> driver.migrationState().get(10, TimeUnit.SECONDS).equals(MigrationState.DUAL_WRITE),
+            "Waiting for KRaftMigrationDriver to enter DUAL_WRITE state");
+
+        Assertions.assertEquals(1, metadataPropagator.images);
+        Assertions.assertEquals(0, metadataPropagator.deltas);
+
+        delta = new MetadataDelta(image);
+        delta.replay(new ConfigRecord()
+            .setResourceType(ConfigResource.Type.BROKER.id())
+            .setResourceName("1")
+            .setName("foo")
+            .setValue("bar"));
+        provenance = new MetadataProvenance(120, 1, 2);
+        image = delta.apply(provenance);
+        enqueueMetadataChangeEventWithFuture(driver, delta, image, provenance).get(10, TimeUnit.SECONDS);
```

Comment:
maybe 1 minute as a timeout? The Jenkins environment can get wacky so short timeouts are almost never a good idea.